fraud/registry.go

```diff
@@ -9,9 +10,16 @@ var (
 	defaultUnmarshalers = map[ProofType]ProofUnmarshaler{}
 )
 
-// Register sets unmarshaler in map by provided ProofType.
-func Register(proofType ProofType, unmarshaler ProofUnmarshaler) {
+// Register adds a string representation and unmarshaller for the provided ProofType.
+func Register(p Proof) {
 	unmarshalersLk.Lock()
 	defer unmarshalersLk.Unlock()
-	defaultUnmarshalers[proofType] = unmarshaler
+	defaultUnmarshalers[p.Type()] = func(data []byte) (Proof, error) {
```

Comment:
We should still panic if the Unmarshaler is already registered. We must not allow overriding